### PR TITLE
4pm-2 - kt - psid sections endpoint

### PR DESCRIPTION
**Pull request by kthanigaivelan**
In this PR, we add the endpoint to get section information given a personal schedule id.
Note: This PR is an update of PR #53  , since I had to use a new branch to get the updates done in main.
Closes Issue #22 

Endpoint: /api/personalSections/all
image (4)

Test plan:
1. Create/ensure that a personal schedule exists. Note the id.
2. Create/ensure there are courses associated with the personal schedule from earlier (the psid field of courses references the id field of personal schedules).
3. The personal sections endpoint, when given the personal schedule id, should output the courses associated with the schedule, formatted in section layout defined by the Course documents.

### CODE REVIEW

**Review by archananeupane — APPROVED** [review]
LGTM!

File: src/main/java/edu/ucsb/cs156/courses/services/UCSBCurriculumService.java [hunk 1]
```diff
@@ -184,4 +186,38 @@ public String getSection(String enrollCode, String quarter) {
...
+        try {
+            ResponseEntity<String> re = restTemplate.exchange(url, HttpMethod.GET, entity, String.class);
+            contentType = re.getHeaders().getContentType();
+            statusCode = re.getStatusCode();
+            retVal = re.getBody();
+        } catch (HttpClientErrorException e) {
+            retVal = "{\"error\": \"401: Unauthorized\"}";
+        }
+        logger.info("json: {} contentType: {} statusCode: {}", retVal, contentType, statusCode);
+        //System.out.println("url=" + url);
```
Comment (andrewhlu) [inline src/main/java/edu/ucsb/cs156/courses/services/UCSBCurriculumService.java:218]: this comment seems unnecessary, can we remove it?

File: src/test/java/edu/ucsb/cs156/courses/controllers/PersonalSectionsControllerTests.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
+package edu.ucsb.cs156.courses.controllers;
+
+import edu.ucsb.cs156.courses.documents.Course;
+import edu.ucsb.cs156.courses.services.UCSBCurriculumService;
+import edu.ucsb.cs156.courses.repositories.UserRepository;
+import edu.ucsb.cs156.courses.testconfig.TestConfig;
+import edu.ucsb.cs156.courses.errors.EntityNotFoundException;
+//import edu.ucsb.cs156.courses.testconfig.SecurityConfig;
```
Comment (andrewhlu) [inline src/test/java/edu/ucsb/cs156/courses/controllers/PersonalSectionsControllerTests.java:8]: ^^ see above

File: src/test/java/edu/ucsb/cs156/courses/controllers/PersonalSectionsControllerTests.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+import java.util.Map;
+import java.util.Optional;
+import java.lang.Iterable;
+import java.lang.String;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
```
Comment (andrewhlu) [inline src/test/java/edu/ucsb/cs156/courses/controllers/PersonalSectionsControllerTests.java:47]: please alphabetize and group similar imports if possible! this makes the import list much easier to read

File: src/test/java/edu/ucsb/cs156/courses/controllers/PersonalSectionsControllerTests.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    @WithMockUser(roles = { "USER" })
+    @Test
+    public void api_psid_sections__user_logged_in__no_personal_schedule() throws Exception {
+
+        MvcResult response = mockMvc.perform(get("/api/personalSections/all?psId=13"))
+                .andExpect(status().is(404)).andReturn();
+        String actual = response.getResponse().getContentAsString();
+        //String expectedResponse = "{\"type\":\"EntityNotFoundException\"}";
+        boolean correct = actual.contains("EntityNotFoundException");
+        //String expectedResponse = "{\"type\":\"EntityNotFoundException\",\"message\":\"PersonalSchedule with id 13 not found\"}";
```
Comment (andrewhlu) [inline src/test/java/edu/ucsb/cs156/courses/controllers/PersonalSectionsControllerTests.java:100]: please remove commented code!

**Review by andrewhlu — CHANGES_REQUESTED** [review]

File: src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSectionsController.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    @ApiOperation(value = "List all sections given a psId")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping(value = "/all", produces = "application/json")
+    public ArrayList<Course> getSectionsByPsId(@ApiParam("psId") @RequestParam Long psId) throws JsonProcessingException{
+        User us = getCurrentUser().getUser();
+        PersonalSchedule ps = personalScheduleRepository.findByIdAndUser(psId,us)
+                .orElseThrow(() -> new EntityNotFoundException(PersonalSchedule.class, psId));
+        ArrayList<Course> sections = new ArrayList<Course>();
+        ArrayList<String> jsons = new ArrayList<String>();
+        Iterable<Courses> courses = coursesRepository.findAllByPsId(psId); //.orElseThrow(() -> new EntityNotFoundException(Courses.class,"psId", psId ));
```
[the remark this answers is not in the document]
Comment (kthanigaivelan) [reply src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSectionsController.java]: Nope, I just removed it!

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**